### ISSUE TITLE
Banning users that left the server.

### ISSUE THREAD

**Issue by MarioPL98** [mumble-voip/mumble#5895]
### Context

Banning users that are no longer connected to the server.

### Description

Currently Mumble has no way to ban users that have left the server. Client should locally store all required info of users that are in chat history (including auto announcements) to allow right click ban option. Currently nothing happens if you right click someone's nickname after the left the server. This should change.

### Mumble component

Both

### OS-specific?

No

**Comment by BryantArias** [comment]
I'm an undergrad student looking to contribute to my first open source project. Would I be able to try implementing this?

**Comment by Krzmbrzl** [comment]
Yes, I would think so. i was thinking that essentially the client needs to keep a liftoff recently disconnected users and use that to provide the required functionality. 
Do you want me to give you some pointers as to where the relevant code parts are located?

**Comment by BryantArias** [comment]
@Krzmbrzl Yes, I would appreciate that. Thank you!

**Comment by Krzmbrzl** [comment]
Alright, so here's what I think might be important in this context:

1. Here's where the context menu gets created when you right-click in the text chat ("log") area:
https://github.com/mumble-voip/mumble/blob/600ffb476a62901d374926cf8cab6875f4c93bbf/src/mumble/MainWindow.cpp#L927-L968
Note in particular
https://github.com/mumble-voip/mumble/blob/600ffb476a62901d374926cf8cab6875f4c93bbf/src/mumble/MainWindow.cpp#L831-L857
which seems to handle right-clicking on a username.
2. Here's the part of the code that gets called when a new user connects to the same server:
https://github.com/mumble-voip/mumble/blob/600ffb476a62901d374926cf8cab6875f4c93bbf/src/mumble/Messages.cpp#L348-L376
and the code that's called when a user disconnects again:
https://github.com/mumble-voip/mumble/blob/600ffb476a62901d374926cf8cab6875f4c93bbf/src/mumble/Messages.cpp#L820-L868

In case it might be relevant/interesting: Here's the function that formats a user's name such that it becomes clickable and interactable in the first place:
https://github.com/mumble-voip/mumble/blob/600ffb476a62901d374926cf8cab6875f4c93bbf/src/mumble/Log.cpp#L531-L556

----

I would suggest you create a draft PR as soon as possible, so I can have a few glances at what direction you are picking and potentially give some advice early on :)

**Comment by fuushyn** [comment]
Hi, is this issue open? I'd like to work on it

**Comment by davidebeatrici** [comment]
Hi, it's indeed still open and as far as I know nobody is working on it currently.

**Comment by Krzmbrzl** [comment]
Actually, @BryantArias wanted to work on this. I have not heard back from them since then though, so I'm not quite sure if that is still the case.

@BryantArias maybe you could comment on this?

**Comment by BryantArias** [comment]
Yes, unfortunately I had to drop this issue and forgot to update you. They're good to take it on if they'd like.

**Comment by fuushyn** [comment]
Cool

**Comment by Krzmbrzl** [comment]
Alright, @fuushyn then this is all yours :)

**Comment by bruhjabi** [comment]
Hello, not sure if @fuushyn ever did anything but I think I could tackle this

**Comment by Hartmnt** [comment]
> Hello, not sure if [@fuushyn](https://github.com/fuushyn) ever did anything but I think I could tackle this

Sure, go ahead! Let us know if you need more info
